source/extensions/filters/network/http_connection_manager/dependency_manager.h

```diff
@@ -0,0 +1,48 @@
+#include "envoy/extensions/filters/common/dependency/v3/dependency.pb.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace HttpConnectionManager {
+
+/**
+ * Validation for an http filter chain based on the factory-level
+ * FilterDependencies specification. @see NamedHttpFilterConfigFactory
+ * Currently only validates decode dependencies.
+ */
+class DependencyManager {
+public:
+  DependencyManager() = default;
+
+  /**
+   * Register each filter in an http filter chain, using name and dependencies
+   * from the filter factory. Filters must be registered in decode path order.
+   */
+  void registerFilter(
+      absl::string_view name,
+      const envoy::extensions::filters::common::dependency::v3::FilterDependencies& dependencies) {
+    filter_chain_.push_back({name, dependencies});
+  }
+
+  /**
+   * Returns true if the decode path of the filter chain is valid. A filter
+   * chain is valid iff for each filter, every decode dependency has been
+   * provided by a previous filter.
+   *
+   * TODO(auni53): Change this to a general valid() that checks decode and
+   * encode path.
+   */
+  bool validDecodeDependencies();
+
+private:
+  std::vector<std::pair<absl::string_view,
```

Comment:
For safety I'd make this an `std::string`. What I mentioned before was about parameters, the persistent structure should avoid holding a string view.

Reply:
Yeah the polymorphism is not working here:
```
dependency_manager.h:26:19: error: no matching member function for call to 'push_back'
    filter_chain_.push_back({name, dependencies});
```

Probably I'd have to call whatever function copies the underlying string from `absl::string_view`, so I just changed it to `const std::string&` both in the function signature and the vector definition.